### ISSUE TITLE
about customdataset

### ISSUE THREAD

**Issue by XLiedInfi** [exiawsh/StreamPETR#178]
hello,
i use my dataset to train the streampetr. what do u think about that, 500 frames is enough to let the transformer convergent or not. the traditional anchor based method is good, but i dont use transformer based approach yet. the scenes is same in the dataset. its a fixes lidars and cameras setting on the infrastructures.

**Comment by exiawsh** [comment]
500 frames is not enough. Especially for PETR. You can try RepDetr3D, which would be better.

**Comment by XLiedInfi** [comment]
@exiawsh i use both point cloud and camera feature. is also not enough right? can u tell me why repdetr3d is better in small dataset size. because of deformableattention?

**Comment by exiawsh** [comment]
Yes. PETR needs more data to learn the 2D-to-3D projection. While deformable attention can do it.

**Comment by reynerliu** [comment]
@XLiedInfi I wonder know which type you tranformed your customdataset to,is it Kitti? I have not find any tools to transfer my own dataset to nuscenes, thus i processed it as kitti form. but after i changed the config profile and began to train, the print info showed grad_norm=nan and loss=0,I don't know why.If your data form is kitti and you succeeded in running,can you share me the train config and dataset config with me?I would be appreciate!
![Uploading IMG_4014.JPG…]()

**Comment by ymqqqqqq** [comment]
@exiawsh  Sorry to bother you. My dataset has 5000 frames, is this enough for PETR to train?

**Comment by exiawsh** [comment]
@ymqqqqqq 
5000 frames might be worth a try. The network should converge. But it may not be very precise in detection.
